base/associative.jl

```diff
@@ -365,6 +367,19 @@ function filter!(f, d::Associative)
     return d
 end
 
+function filter_in_one_pass!(f, d::Associative)
+    try
+        for pair in d
+            if !f(pair)
+                delete!(d, pair.first)
+            end
+        end
+    catch e
+        return filter!_dict_deprecation(e, f, d)
```

Comment:
Why is this deprecated still? Wasn’t this a long time ago?

Reply:
Nope, this is a 0.7 deprecation. Maybe the 0.7 cycle just feels like a long time :)